src/Application/src/CalculatorApplication.cpp

```diff
@@ -6,24 +6,28 @@
  */
 ////////////////////////////////////////////////////////////////////////////
 
+#include <chrono>
+#include <ctime>
 #include <future>
 #include <iostream>
 #include <limits>
 #include <sstream>
 #include <string>
 #include <thread>
 
+#include "AddCalculator.h"
 #include "CalculatorApplication.h"
 #include "CalculatorApplicationFactory.h"
 #include "CalculatorStrings.h"
+#include "History.h"
 
 namespace calculator {
 
 // *****************************************************************************/
 // ***************** CalculatorApplication public methods **********************/
 // *****************************************************************************/
 
-int CalculatorApplication::calculate(float firstNumber, float secondNumber, char operation)
+int CalculatorApplication::calculate(float firstNumber, float secondNumber, char operation, std::string originalEquation)
```

Comment:
When a function starts getting quite a few arguments, It might be better to pass an object with all that info instead.